pushdeploy/apiv1.py

```diff
@@ -25,26 +12,71 @@ def init_api():
     """Creates instances of the incluster config and client API
     and stores them in global"""
     g.configuration = config.load_incluster_config()
-    g.api_instance = client.AppsV1Api(client.ApiClient(g.configuration))
+    g.apps_v1_api_instance = client.AppsV1Api(client.ApiClient(g.configuration))
+    g.batch_v1beta1_instance = client.BatchV1beta1Api(client.ApiClient(g.configuration))
     g.PD_REGISTRY = current_app.config['PD_REGISTRY']
 
-def read_deployment(deployment, namespace):
+def list_cron_job(name, namespace):
```

Comment:
All of these `list_*` methods seem to follow the same pattern.

I could see a more general `fetch_object(object, name, namespace)` (something like that) that would dispatch the calls to the necessary k8s api endpoint. At that point, you could either use the more generic `fetch_object` or keep the more specialized calls that :just: use the generic one under the hood.